### ISSUE TITLE
[Windows] Ctrl + Backspace does not work properly in text fields

### ISSUE THREAD

**Issue by craigcarr** [flutter/flutter#90427]
## Steps to Reproduce

Create a Flutter app running on Windows with a `TextField`. Type in some words. Then use the `Ctrl + Backspace` keyboard shortcut to delete a word. The following will happen:

1. The "box" character will appear instead of deleting the previous word.
2. If you use `Ctrl + Backspace` again, then the previous word will be deleted, but the box character will remain.
3. If you use `Ctrl + Backspace` yet again, the word will re-appear and the box character will still remain.

https://user-images.githubusercontent.com/13004183/134093415-5297fb8c-f5e4-47c8-9373-b846df7a5e41.mp4

This issue seems similar to https://github.com/flutter/flutter/issues/78974, except the issue has been closed and supposedly fixed by https://github.com/flutter/flutter/pull/79695, but it does not appear to be fixed here.

`flutter doctor -v` output:

```
[√] Flutter (Channel stable, 2.5.0, on Microsoft Windows [Version 10.0.19043.1237], locale en-US)
    • Flutter version 2.5.0 at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (13 days ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Build Tools 2019 16.8.3)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
    • Visual Studio Build Tools 2019 version 16.8.30804.86
    • Windows 10 SDK version 10.0.18362.0

[√] VS Code (version 1.60.1)
    • VS Code at C:\Users\chaox\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.26.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1237]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 93.0.4577.82
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 93.0.961.52
```

**Comment by darshankawar** [comment]
Verified this issue on latest stable (2.5.1) and on latest master (2.6.0-6.0.pre.160).  The issue doesn't replicate on master, as, using `Ctrl + backspace` deletes the text properly as expected:

![master_works](https://user-images.githubusercontent.com/67046386/134137052-6cf58132-5ded-4eed-bc9b-777dae140c2f.png)

Although, the change doesn't seem to have landed in stable, as I see the block as below while trying to delete using `Ctrl+backspace`:

![ctrl+bck](https://user-images.githubusercontent.com/67046386/134137168-ab7df5fa-4c42-4b0f-a1c5-8cae8d1d8598.png)

@craigcarr You may switch to latest master to see it working properly and may have to wait till this change rolls in stable.

<details>
<summary> flutter doctor -v </summary>

```

[√] Flutter (Channel master, 2.6.0-6.0.pre.160, on Microsoft Windows [Version 10.0.19042.1237], locale en-US)
    • Flutter version 2.6.0-6.0.pre.160 at D:\FlutterSDK\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a133306791 (7 hours ago), 2021-09-20 19:53:05 -0400
    • Engine revision 7e94275f73
    • Dart version 2.15.0 (build 2.15.0-126.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\Win\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.60.1)
    • VS Code at C:\Users\Win\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1237]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 93.0.4577.82
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 93.0.961.52

• No issues found!


[√] Flutter (Channel stable, 2.5.1, on Microsoft Windows [Version 10.0.19042.1237], locale en-US)
    • Flutter version 2.5.1 at D:\FlutterSDK\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (4 days ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\Win\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.60.1)
    • VS Code at C:\Users\Win\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (4 available)
    • SM A260G (mobile) • 5200763ebcfa861f • android-arm    • Android 8.1.0 (API 27)
    • Windows (desktop) • windows          • windows-x64    • Microsoft Windows [Version 10.0.19042.1237]
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 93.0.4577.82
    • Edge (web)        • edge             • web-javascript • Microsoft Edge 93.0.961.52

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
I am going ahead and closing this as already fixed in master.

@gspencergoog @justinmc Just would like to know if this issue can be cherrypicked for 2.5 ? Since this was fixed recently but doesn't seem to have landed in stable. https://github.com/flutter/flutter/issues/78974

**Comment by justinmc** [comment]
@darshankawar Do you know which PR fixed it or should I try bisecting it?  If it wasn't a huge PR like https://github.com/flutter/flutter/pull/86736 then should be fine to cherry pick.

**Comment by darshankawar** [comment]
@justinmc 
I believe [this](https://github.com/flutter/flutter/pull/79695) was the PR that introduced this fix. Per that PR:

<img width="410" alt="Screenshot 2021-09-22 at 11 03 14 AM" src="https://user-images.githubusercontent.com/67046386/134288535-8d9bac91-f80b-4be0-a00a-af37053e92fa.png">

If this is correct, I looked at the 2.5 release [notes](https://flutter.dev/docs/development/tools/sdk/release-notes/release-notes-2.5.0) and didn't find the said PR in the list.

**Comment by darshankawar** [comment]
@justinmc added `cp: 2.5` label.

**Comment by devoncarew** [comment]
@justinmc @darshankawar - can you double check your cherry pick request? It does look like the commit for PR https://github.com/flutter/flutter/pull/79695 is in 2.5.0 / 2.5.1:

https://github.com/flutter/flutter/commit/0ba2f6a87fbfd857c0a4e01a47e824dcab5462de

If you go to the list of tags for that commit (near the top of the page), you can expand the list by clicking on the '...' label.

**Comment by justinmc** [comment]
I think you're right, I'm removing the `cp` label but please re-add it if anyone thinks I'm wrong.

On my Windows machine I checked out ffb2ecea5223acdd139a5039be2f9c796962833d, which is the commit tagged as 2.5.1, and ctrl + backspace seemed to work fine.

Also side note, I was unable to verify that https://github.com/flutter/flutter/pull/79695 was the PR that fixed this. 
 When I check out the commit before it (0ba2f6a87fbfd857c0a4e01a47e824dcab5462de^), then it gives me an error when I try to run it on Windows...  But anyway it seems like the fix is in 2.5.1 and we're good.
